components/ca-crud.js

```diff
@@ -0,0 +1,810 @@
+define([
+    './helpers/component-support.js',
+    './helpers/clear-element.js',
+    './helpers/secureajax.js',
+    './helpers/trace.js',
+    './helpers/create-element.js',
+    './helpers/schema-to-default.js',
+    './helpers/to-class-name.js',
+    './helpers/dialogs.js',
+    './helpers/fire-event.js',
+    './helpers/clone-object.js',
+    './helpers/schema-actions.js',
+    './helpers/capitalize.js',
+    './helpers/distinct.js',
+    'document-register-element',
+    './ca-form.js',
+    './ca-list.js'
+], (componentSupport, clearElement, ajax, trace, createElement, schemaToDefault, toClassName, dialogs, fireEvent, cloneObject, schemaActions, capitalize, distinct) => {
+    /**
+     * @exports ca-crud
+     * A custom HTML element (Web Component) that can be created using document.createElement('ca-crud') or included in a HTML page as an element.
+     */
+    class Crud extends HTMLElement {
```

Comment:
I think the convention is to use CaCrud here, otherwise we'll get to Select and RadioGroup and have to use Ca to avoid naming conflicts, and then be inconsistent.